### DISCUSSION TITLE
Cannot build a project with wagmi generated code

### DISCUSSION THREAD

**Discussion by kostysh** [wevm/wagmi#2415] [General]
Trying to build a Typescript project with the code that generated using the `@wagmi/cli` but getting 95 compiler errors like:

```
generated.ts:3703:17 - error TS2742: The inferred type of 'useMarketCheckOut' cannot be named without a reference to '../node_modules/abitype/dist/abi-3a9c20c7.js'. This is likely not portable. A type annotation is necessary.
```

Here is my `tsconfig.json`:

```json
{
  "compilerOptions": {
    "strict": true,
    "esModuleInterop": true,
    "lib": ["ESNext", "DOM", "DOM.Iterable"],
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "NodeNext",
    "skipLibCheck": true,
    "checkJs": true,
    "allowJs": true,
    "declaration": true,
    "declarationMap": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "noErrorTruncation": true,
    "forceConsistentCasingInFileNames": true,
    "useUnknownInCatchVariables": true,
    "importsNotUsedAsValues": "remove",
    "noImplicitReturns": true
  },
  "include": ["./src/index.ts"]
}
```

What I am doing wrong?

My environment:

- typescript: 5.0.4
- wagmi: 1.0.5
- @wagmi/cli: 1.0.1

**Comment by jxom** [comment]
Any chance you could share your repo or another minimal reproducible example so we can debug?

**Reply by kostysh** [reply]
> Any chance you could share your repo or another minimal reproducible example so we can debug?

I have created this example that reproduces the issue: https://stackblitz.com/edit/typescript-f3kfuh?file=package.json

You can download this shared project, to start:

```bash
yarn
yarn build
```

**Comment by kasparkallas** [comment]
I personally solved this by only generating the ABIs and using the regular wagmi hooks.

Just my 2 cents.
